includes/Core/Assets/Assets.php

```diff
@@ -766,8 +766,21 @@ private function get_inline_user_data() {
 	 * @return array The dashboard sharing inline data to be output.
 	 */
 	private function get_inline_dashboard_sharing_data() {
+		global $wp_roles;
+
+		$all_roles   = wp_roles()->roles;
 		$inline_data = array();
 
+		foreach ( $all_roles as $role => $details ) {
+			// Filter only the roles with the capability `edit_posts`.
+			$role_to_filter = get_role( $role );
+			if ( $role_to_filter->has_cap( 'edit_posts' ) ) {
+				$sub['id']          = esc_attr( $role );
+				$sub['displayName'] = translate_user_role( $details['name'] );
```

Comment:
Implicit array declaration is a bad habit. A variable needs to be declared as an array before you can assign a value to its property.